subsys/net/lib/nrf_cloud/src/nrf_cloud_client_id.c

```diff
@@ -63,22 +63,10 @@ int nrf_cloud_configured_client_id_get(char * const buf, const size_t buf_sz)
 	int err;
 	int print_ret;
 
-#if (defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_INTERNAL_UUID) || \
-	defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI))
-	/* UUID/IMEI are obtained via AT command */
-	if (!IS_ENABLED(CONFIG_AT_CMD_SYS_INIT)) {
-		err = at_cmd_init();
-		if (err) {
-			LOG_ERR("at_cmd failed to initialize, error: %d", err);
-			return err;
-		}
-	}
-#endif
-
 #if defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI)
-	char imei_buf[CGSN_RESPONSE_LENGTH + 1];
+	char imei_buf[CGSN_RESPONSE_LENGTH];
 
-	err = at_cmd_write("AT+CGSN", imei_buf, sizeof(imei_buf), NULL);
+	err = nrf_modem_at_cmd(imei_buf, sizeof(imei_buf), "AT+CGSN");
 	if (err) {
 		LOG_ERR("Failed to obtain IMEI, error: %d", err);
 		return err;
```

Comment:
I cannot comment few lines below so adding here:
`imei_buf[NRF_IMEI_LEN] = 0;`

I guess this adds NULL-termination to a string. Would '\0' be more common way of doing that?

Reply:
i does indeed add null termination.
i do not know what the most common way is.